airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.kt

```diff
@@ -573,39 +583,63 @@ class SnowflakeDestinationHandler(
-            val existingTables = LinkedHashMap<String, LinkedHashMap<String, TableDefinition>>()
-            // convert list stream to array
-            val namespaces = streamIds.map { it.finalNamespace }.toTypedArray()
-            val names = streamIds.map { it.finalName }.toTypedArray()
-            val query =
-                """
-                |SELECT table_schema, table_name, column_name, data_type, is_nullable 
-                |FROM information_schema.columns 
-                |WHERE table_catalog = ? 
-                |AND table_schema IN (${IntRange(1, streamIds.size).joinToString { "?" }}) 
-                |AND table_name IN (${IntRange(1, streamIds.size).joinToString { "?" }}) 
-                |ORDER BY table_schema, table_name, ordinal_position; 
-                |""".trimMargin()
-
-            val bindValues =
-                arrayOf(databaseName.uppercase(Locale.getDefault())) + namespaces + names
-            val results: List<JsonNode> = database.queryJsons(query, *bindValues)
-            for (result in results) {
-                val tableSchema = result["TABLE_SCHEMA"].asText()
-                val tableName = result["TABLE_NAME"].asText()
-                val columnName = result["COLUMN_NAME"].asText()
-                val dataType = result["DATA_TYPE"].asText()
-                val isNullable = result["IS_NULLABLE"].asText()
-                val tableDefinition =
-                    existingTables
-                        .computeIfAbsent(tableSchema) { _: String? -> LinkedHashMap() }
-                        .computeIfAbsent(tableName) { _: String? ->
-                            TableDefinition(LinkedHashMap())
+
+            val existingTablesFromShowQuery =
+                LinkedHashMap<String, LinkedHashMap<String, TableDefinition>>()
+            var showColumnsResult: List<JsonNode> = listOf()
+
+            try {
+                for (stream in streamIds) {
+                    val showColumnsQuery =
+                        String.format(
+                            """
+                                SHOW COLUMNS IN TABLE "%s"."%s"."%s";
+                            """.trimIndent(),
+                            databaseName,
+                            stream.finalNamespace,
+                            stream.finalName,
+                        )
+                    showColumnsResult = database.queryJsons(
+                        showColumnsQuery,
+                    )
+
+                    for (result in showColumnsResult) {
+                        val tableSchema = result["schema_name"].asText()
+                        val tableName = result["table_name"].asText()
+                        val columnName = result["column_name"].asText()
+                        var dataType = JSONObject(result["data_type"].asText()).getString("type")
+
+                        //TODO: Need to check if there are other datatype differences
```

Comment:
that should probably be checked before merging this PR. Do we have any automated test that checks this?

Reply:
per the doc, it seems those are the only differences. I'd rather have a comment that points to [https://docs.snowflake.com/en/sql-reference/sql/show-columns#output](https://docs.snowflake.com/en/sql-reference/sql/show-columns#output) rather than a `TODO`